<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Peter Dimler" />

<meta name="date" content="2023-09-10" />

<title>Stroke Analysis</title>

<script src="site_libs/header-attrs-2.18/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html"></a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="ICR_Project.html">ICR Analysis</a>
</li>
<li>
  <a href="EDAmigo.html">EDAmigo</a>
</li>
<li>
  <a href="STROKE_SCRUBBED.html">Stroke Analysis</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Stroke Analysis</h1>
<h4 class="author">Peter Dimler</h4>
<h4 class="date">2023-09-10</h4>

</div>


<div id="executive-summary" class="section level1">
<h1>Executive Summary</h1>
<p>Between February of 2021 and April of 2022, a <a
href="https://www.penumbrainc.com/">Penumbra</a>-sponsored study
collected medical data, encompassing social history to genetics from 400
different stroke patients. The goal of the study was to discover
medically relevant details about patients that could further the
understanding of stroke severity and treatment. This analysis explored
over 10GB of data and found evidence that helps explain the variability
in stroke severity among male and female patients.</p>
</div>
<div id="project-overview" class="section level1">
<h1>Project Overview</h1>
<div id="background" class="section level2">
<h2>Background</h2>
<p>Stroke is a <a
href="https://www.cdc.gov/stroke/facts.htm#:~:text=Every%2040%20seconds%2C%20someone%20in%20the%20United%20States%20has%20a%20stroke.&amp;text=Every%203%20minutes%20and%2014%20seconds%2C%20someone%20dies%20of%20stroke.&amp;text=Every%20year%2C%20more%20than%20795%2C000,United%20States%20have%20a%20stroke.">leading
cause of long term disability</a> in the United States. According to the
CDC, more than <strong>795,000</strong> Americans suffer from a stroke
each year. A stroke is a medical emergency where either:</p>
<ul>
<li>blood flow to the brain is blocked by a clot (Ischemic Stroke)
or</li>
<li>bleeding in the brain occurs due to a ruptured artery (Hemorrhagic
Stroke)</li>
</ul>
<p>Both events can damage the brain and cause long-term disability or
death. The severity of a stroke is often measured using the National
Institutes of Health Stroke Scale (<a
href="https://www.physio-pedia.com/NIH_Stroke_Scale">NIHSS</a>). The
final score (ranging from 0 to 42) is derived using 15 neurological
examination questions and the stroke severity can roughly be interpreted
using the following bands:</p>
<ul>
<li>Very Severe: &gt;= 25</li>
<li>Severe: 15 - 24</li>
<li>Moderate: 5 - 14</li>
<li>Mild: 1 - 4</li>
</ul>
<p>Between February of 2021 and April of 2022 various medical details
were collected from 400 stroke patients as part of a study sponsored by
<a href="https://www.penumbrainc.com/">Penumbra</a>, a company that is
developing products to help treat stroke patients. The data is broken
into the following three categories:</p>
<ul>
<li>Hospital Data: The study attempted to collect each patient’s medical
and social history as well as specific details related to their stroke
and any treatment they received. The data is gathered into a table that
has 400 rows (1 for each patient) and 248 columns which capture the
details gathered for each patient.</li>
<li>Proteomics: The study collected blood and clot samples from 80 of
the patients and Creative Proteomics ran the protein samples on SDS-PAGE
gel followed by in-gel digestion and then identified and quantified
1600+ proteins by applying their nanoLC-MS/MS platform.</li>
<li>Genetics: Genotypes for each of the 400 patients were extracted in
the form of Single Polynucleotide Polymorphisms (SNPs). Each SNP is a
single base location in the DNA where there is known to be substantial
population variability. The SNP data collected in the study has over
654K SNPs totaling 9.97GB of data.</li>
</ul>
</div>
<div id="objective" class="section level2">
<h2>Objective</h2>
<p>The goal of this analysis was to identify any information contained
in the study that could advance our understanding of strokes with the
hope of improving patient treatment. Accordingly, this analysis focused
on creating explanatory models as opposed to a predictive model.</p>
</div>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="hospital-data" class="section level2">
<h2>Hospital Data</h2>
<div id="exploratory-data-analysis-eda" class="section level3">
<h3>Exploratory Data Analysis (EDA)</h3>
<p>Since the objective of this analysis was to construct an explanatory
model, EDA played a key role in assessing data quality, choosing a
response variable, selecting candidate predictors, and identifying any
data handling techniques needed.</p>
<div id="choosing-a-response" class="section level4">
<h4>Choosing a Response</h4>
<p>The hospital data had several variables that could be used to assess
the severity of a patient’s stroke. Such variables include the scores
assigned using the Glasgow Coma Scale (GCSSCTOT), National Institutes of
Health Stroke Scale (NIHSSTOT), and Modified Rankin Scale (MRSSCORE).
Changes in these scores between patient admission and discharge could
also be considered to search for treatment effects. Choosing which
variable to use as the response for this analysis boiled down to
selecting the one with the fewest missing values. For admitted patients,
the GCSSCTOT was missing for 335 of the 400 patients, the MRSSCORE was
missing for 86 of the patients, and the NIHSSTOT score was only missing
for 2 of the patients. Missing values were prevalent within the
discharge details as well and as a result, the patient’s total National
Institutes of Health Stroke Scale score (NIHSSTOT) at admission was
selected as the response variable for this analysis.</p>
<p>Univariate statistics for the NIHSSTOT value can be viewed by
expanding the drop down below. The first 2 plots show the distribution
of the NIHSSTOT values and the bottom plot shows its cumulative
distribution. From these plots we can see that our patient’s stroke
severity scores are slightly positively skewed and that they cover the
full range of scores with half of the patients having Mild to Moderate
stroke severity scores and half having Severe to Very Severe scores.</p>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/NIHSSTOT_DESC.png">
</details>
</div>
<div id="selecting-candidate-predictors" class="section level4">
<h4>Selecting Candidate Predictors</h4>
<p>Since the goal of this analysis was to create an explanatory model,
candidate selection for potential predictors was done by manually
reviewing the 240+ predictors to determine if there was enough data
present to merit an analysis, if there were potential differences in
centrality and spread, and if the predictor should reasonably be
included in the model based on previous stroke analyses or demographic
areas of interest. In the end, the following 23 predictors were selected
as candidates for an explanatory model.</p>
<ol style="list-style-type: decimal">
<li>GLUC: Glucose Level (mg/dL)</li>
<li>WBC: White Blood Cell Count (k/uL)</li>
<li>RBC: Red Blood Cell Sount (k/uL)</li>
<li>HCT: Hematocrit (Percentage of red blood cells by volume)</li>
<li>HBG: Hemoglobin (g/dL)</li>
<li>CLCLTAR: Clot Area</li>
<li>CLCLTWT: Clot Weight</li>
<li>AGE: Age in years (integer)</li>
<li>SEX: Sex male (1) or female (2)</li>
<li>HEIGHT: Height (CM)</li>
<li>WEIGHT: Weight (KG)</li>
<li>BMI: Body Mass Index (kg/m^2)</li>
<li>MHNONE: Pertinent medical history (Y = Yes)</li>
<li>MHPSIS: Suffered previous ischemic stroke (Y = Yes)</li>
<li>MHPSTIA: Suffered previous transient ischemic attack (Y = Yes)</li>
<li>MHDVT: Suffered from deep vein thrombosis (Y = Yes)</li>
<li>MHDM: Has Diabetes ( Y = Yes)</li>
<li>MHHTN: Has Hypertension (Y = Yes)</li>
<li>MHTHROMB: Has medical history of Thrombocytopenia</li>
<li>MHATEXCR: Has medical history of Extracranial - Carotid
Atherosclerosis</li>
<li>MHPSISEL: Previous type of ischemic stroke (when applicable)</li>
<li>SHALCUSE: History of Alcohol Use</li>
<li>SHMRJYN: History of Marijuana use</li>
</ol>
<p>The summary statistics for each candidate predictor are provided in
the tabs below and can be viewed by selecting the predictor tab and
clicking the univariate statistics drop down arrow. The relationship
between each candidate predictor and the response variable (NIHSSTOT)
can be reviewed by expanding the bivariate statistics drop down.</p>
</div>
<div id="numeric-candidates" class="section level4 tabset">
<h4 class="tabset">Numeric Candidates</h4>
<div id="gluc" class="section level5">
<h5>GLUC</h5>
<ul>
<li><strong>Category:</strong> Baseline Laboratory Values</li>
<li><strong>Name:</strong> Glucose</li>
<li><strong>Label:</strong> GLUC</li>
<li><strong>Details:</strong> The glucose levels of all of the patients
in this study were measured in mg/dL. Notably, the data is positively
skewed and has a slight positive correlation with stroke severity.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_GLUC_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_GLUC_BV_DESC.png">
</details>
</div>
<div id="wbc" class="section level5">
<h5>WBC</h5>
<ul>
<li><strong>Category:</strong> Baseline Laboratory Values</li>
<li><strong>Name:</strong> White Blood Cells</li>
<li><strong>Label:</strong> WBC</li>
<li><strong>Details:</strong> The white blood cell count was recorded in
k/uL or 10^3 cells /mm^3 for nearly every patient. The units are
equivalent and the standard range for white blood cell count count is 4
- 11 K/uL. One notable feature of the data is that it is positively
skewed and has a slight positive correlation with stroke severity.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_WBC_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_WBC_BV_DESC.png">
</details>
</div>
<div id="rbc" class="section level5">
<h5>RBC</h5>
<ul>
<li><strong>Category:</strong> Baseline Laboratory Values</li>
<li><strong>Name:</strong> Red Blood Cells</li>
<li><strong>Label:</strong> RBC</li>
<li><strong>Details:</strong> The red blood cell count was recorded
(equivalently) in k/uL or 10^3 cells /mm^3 for nearly every patient. The
data does not have any apparent outliers and is fairly symmetric in
distribution. The red blood cell count, on its own, isn’t correlated
with the stroke severity but was included since it helps describe the
patient’s blood composition.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_RBC_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_RBC_BV_DESC.png">
</details>
</div>
<div id="hct" class="section level5">
<h5>HCT</h5>
<ul>
<li><strong>Category:</strong> Baseline Laboratory Values</li>
<li><strong>Name:</strong> Hematocrit</li>
<li><strong>Label:</strong> HCT</li>
<li><strong>Details:</strong> Hematocrit is the percentage of red blood
cells by volume. The data is fairly symmetric in distribution and
doesn’t appear to have any notable outliers. Like the red blood cell
count, Hematocrit on its own isn’t correlated with the stroke severity
but was included since it helps describe the patient’s blood
composition.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_HCT_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_HCT_BV_DESC.png">
</details>
</div>
<div id="hbg" class="section level5">
<h5>HBG</h5>
<ul>
<li><strong>Category:</strong> Baseline Laboratory Values</li>
<li><strong>Name:</strong> Hemoglobin</li>
<li><strong>Label:</strong> HBG</li>
<li><strong>Details:</strong> Each patient’s hemoglobin level was
measured in g/dl. The data is fairly symmetric in distribution and
doesn’t appear to have any notable outliers. Hemoglobin on its own isn’t
correlated with the stroke severity but was included since it helps
describe the patient’s blood composition.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_HBG_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_HBG_BV_DESC.png">
</details>
</div>
<div id="clcltar" class="section level5">
<h5>CLCLTAR</h5>
<ul>
<li><strong>Category:</strong> Histopathology Results of Thrombus
Retrieval</li>
<li><strong>Name:</strong> Clot area</li>
<li><strong>Label:</strong> CLCLTAR</li>
<li><strong>Details:</strong> The CLCLTAR is the clot area. The units
were not provided but were presumably entered in square millimeters. The
data is positively skewed and does appear to have some outliers.
Additionally, there were 42 patients that did not have a value for the
clot area. The clot area was included since it has a slight positive
correlation with stroke severity.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_CLCLTAR_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_CLCLTAR_BV_DESC.png">
</details>
</div>
<div id="clcltwt" class="section level5">
<h5>CLCLTWT</h5>
<ul>
<li><strong>Category:</strong> Histopathology Results of Thrombus
Retrieval</li>
<li><strong>Name:</strong> Clot weight</li>
<li><strong>Label:</strong> CLCLTWT</li>
<li><strong>Details:</strong> The CLCLTWT is the clot weight. The units
were not provided and the data is positively skewed and does appear to
have some outiers. Similar to CLCLTAR, the clot weight was not available
for 42 of the patients. The clot weight was included since it has a
slight positive correlation with stroke severity.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_CLCLTWT_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_CLCLTWT_BV_DESC.png">
</details>
</div>
<div id="age" class="section level5">
<h5>AGE</h5>
<ul>
<li><strong>Category:</strong> Demographics</li>
<li><strong>Name:</strong> Age</li>
<li><strong>Label:</strong> AGE</li>
<li><strong>Details:</strong> The Age variable, measured in years, is
symmetrically distributed without any outliers. Age, on its own, doesn’t
appear to have a notable correlation with stroke severity but was
included since it helps describe the patient’s demographics.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_AGE_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_AGE_BV_DESC.png">
</details>
</div>
<div id="height" class="section level5">
<h5>HEIGHT</h5>
<ul>
<li><strong>Category:</strong> Demographics</li>
<li><strong>Name:</strong> Height</li>
<li><strong>Label:</strong> HEIGHT</li>
<li><strong>Details:</strong> The patient’s height was measured in
centimeters and the values are fairly symmetric in distribution. Two of
the patients had heights less than 95 centimeters which seemed unlikely
and causes issues with BMI. As a result, the data associated with these
patients was dropped from the analysis. The patient’s height does not
have a notable correlation with the stroke severity but was included
since it helps describe the patient’s physical attributes.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_HEIGHT_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_HEIGHT_BV_DESC.png">
</details>
</div>
<div id="weight" class="section level5">
<h5>WEIGHT</h5>
<ul>
<li><strong>Category:</strong> Demographics</li>
<li><strong>Name:</strong> Weight</li>
<li><strong>Label:</strong> WEIGHT</li>
<li><strong>Details:</strong> Each patient’s weight was measured in
kilograms and the values are slightly positively skewed. The patient’s
weight does not have a notable correlation with the stroke severity but
was included since it helps describe the patient’s physical
attributes.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_WEIGHT_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_WEIGHT_BV_DESC.png">
</details>
</div>
<div id="bmi" class="section level5">
<h5>BMI</h5>
<ul>
<li><strong>Category:</strong> Demographics</li>
<li><strong>Name:</strong> Body Mass Index</li>
<li><strong>Label:</strong> BMI</li>
<li><strong>Details:</strong> BMI is the patient’s Body Mass Index in
kilograms / meter^2. The values are positively skewed and contain 2
outliers as a result of the patients with heights less than 95cm. BMI
does not have a notable correlation with the stroke severity but was
included since it helps describe the patient’s physical attributes.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_BMI_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_BMI_BV_DESC.png">
</details>
</div>
</div>
<div id="categorical-candidates" class="section level4 tabset">
<h4 class="tabset">Categorical Candidates</h4>
<div id="sex" class="section level5">
<h5>SEX</h5>
<ul>
<li><strong>Category:</strong> Demographics</li>
<li><strong>Name:</strong> Sex</li>
<li><strong>Label:</strong> SEX</li>
<li><strong>Details:</strong> The Sex variable indicates if a patient is
a male (1) or female (2). While there isn’t a noticeable difference in
the mean stroke severity, the box plots have a difference in variance
and sex was selected as a candidate variable as a result.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_SEX_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_SEX_BV_DESC.png">
</details>
</div>
<div id="mhnone" class="section level5">
<h5>MHNONE</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> No Pertinent Medical History</li>
<li><strong>Label:</strong> MHNONE</li>
<li><strong>Details:</strong> The MHNONE variable is ‘Y’ if there is no
pertinent medical history for the Patient. Patients who indicated that
they had no pertinent medical history appear (from the boxplots) to have
less severe strokes and the factor was selected as a candidate as a
result.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHNONE_BV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHNONE_UV_DESC.png">
</details>
</div>
<div id="mhpsis" class="section level5">
<h5>MHPSIS</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Previous Ischemic Stroke</li>
<li><strong>Label:</strong> MHPSIS</li>
<li><strong>Details:</strong> The MHPSIS variable indicates if a patient
suffered from a previous ischemic stroke. Patients who suffered from
prior Ischemic strokes appear to experience more severe strokes making
this a good candidate predictor.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHPSIS_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHPSIS_BV_DESC.png">
</details>
</div>
<div id="mhpstia" class="section level5">
<h5>MHPSTIA</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Previous Transient Ischemic Attack</li>
<li><strong>Label:</strong> MHPSTIA</li>
<li><strong>Details:</strong> The MHPSTIA variable indicates if a
patient had a previous transient ischemic attack. Only 10 patients
indicated that they had a previous transient ischemic attack and their
median stroke severity appears (based on the boxplots) to be slightly
less than those who didn’t experience a previous transient ischemic
attack. This was selected as a candidate predictor due to its relation
to ischemic strokes.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHPSTIA_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHPSTIA_BV_DESC.png">
</details>
</div>
<div id="mhdvt" class="section level5">
<h5>MHDVT</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Deep Vein Thrombosis</li>
<li><strong>Label:</strong> MHDVT</li>
<li><strong>Details:</strong> MHDVT indicates if a patient has suffered
from deep vein thrombosis. An interesting feature of the boxplots found
under the Bivariate statistics below is that the data for patients who
suffered from deep vein thrombosis appears to be positively skewed. This
may suggest that these patients are less likely to experience mild to
moderate strokes.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHDVT_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHDVT_BV_DESC.png">
</details>
</div>
<div id="mhdm" class="section level5">
<h5>MHDM</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Diabetes Mellitus</li>
<li><strong>Label:</strong> MHDM</li>
<li><strong>Details:</strong> The MHDM variable indicates if a patient
has diabetes. Patients with diabetes appear to suffer from more severe
strokes on average. This aligns with GLUC numeric predictor.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHDM_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHDM_BV_DESC.png">
</details>
</div>
<div id="mhhtn" class="section level5">
<h5>MHHTN</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Hypertension</li>
<li><strong>Label:</strong> MHHTN</li>
<li><strong>Details:</strong> The MHHTN variable indicates if a patient
has Hypertension (High blood-pressure). Patients with Hypertension
appear to have more severe strokes on average.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHHTN_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHHTN_BV_DESC.png">
</details>
</div>
<div id="mhthromb" class="section level5">
<h5>MHTHROMB</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Thrombocytopenia</li>
<li><strong>Label:</strong> MHTHROMB</li>
<li><strong>Details:</strong> The MHTHROMB variable captures if a
patient has a medical history of Thrombocytopenia which occurs when
platelet counts are low. There seems to be a difference in severity
between patients who have a history of Thrombocytopenia and those who
don’t but there are a lot of missing values.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHTHROMB_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHTHROMB_BV_DESC.png">
</details>
</div>
<div id="mhatexcr" class="section level5">
<h5>MHATEXCR</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Extracranial - Carotid Atherosclerosis</li>
<li><strong>Label:</strong> MHATEXCR</li>
<li><strong>Details:</strong> The MHATEXCR variable captures if a
patient has a medical history of Extracranial - Carotid Atherosclerosis
which is a hardening and narrowing of vessels due to fat deposits. There
seems to be a difference in severity between patients who have a history
of Extracranial - Carotid Atherosclerosis and those who don’t but there
are a lot of missing values.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHATEXCR_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHATEXCR_BV_DESC.png">
</details>
</div>
<div id="mhpsisel" class="section level5">
<h5>MHPSISEL</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Type of Previous Ischemic Stroke</li>
<li><strong>Label:</strong> MHPSISEL</li>
<li><strong>Details:</strong> The MHPSISEL variable captures the
previous type of ischemic stroke (when applicable). While there are
quite a few missing values, it does appear that patients who had a
cardio or cryptogenic Ischemic Stroke previously experience more severe
strokes when compared against patients that didn’t have a previous
ischemic stroke or had a LAA or SAO stroke.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHPSISEL_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_MHPSISEL_BV_DESC.png">
</details>
</div>
<div id="shmrjyn" class="section level5">
<h5>SHMRJYN</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Marijuana Use</li>
<li><strong>Label:</strong> SHMRJYN</li>
<li><strong>Details:</strong> The SHMRJYN variable indicates if a
patient uses marijuana. The data for users appears to be positively
skewed which may be an indication that users are less likely to
experience mild to moderate strokes.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_SHMRJYN_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_SHMRJYN_BV_DESC.png">
</details>
</div>
<div id="shalcuse" class="section level5">
<h5>SHALCUSE</h5>
<ul>
<li><strong>Category:</strong> Medical History</li>
<li><strong>Name:</strong> Frequency of Alcohol Use</li>
<li><strong>Label:</strong> SHALCUSE</li>
<li><strong>Details:</strong> The SHALCUSE variable captures how many
drinks a person estimates that they have per week. Although there are
relatively low counts, it appears that the stroke severity of people who
report drinking weekly is lower than it is for people who don’t.</li>
</ul>
<details>
<summary>
Univariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_SHALCUSE_UV_DESC.png">
</details>
<details>
<summary>
Bivariate Statistics
</summary>
<img src = "Data/GRAPHICS/EDA_SHALCUSE_BV_DESC.png">
</details>
</div>
</div>
</div>
<div id="linear-regression" class="section level3">
<h3>Linear Regression</h3>
<p>Linear regression can be used to create an explanatory model that
helps us understand which of the candidate factors selected in the EDA
process contribute to higher stroke severity scores in patients. As
noted in the EDA section, some of the candidate predictors are highly
skewed, have outliers, and contain missing values all of which can pose
challenges for linear regression. These issues were addressed via data
selection, imputation, and data transformations.</p>
<div id="data-selection" class="section level4">
<h4>Data selection</h4>
<p>The hospital data contains data for patients suffering from two
distinct stroke types, ischemic and hemorrhagic. There were only 10
patients who both suffered from hemorrhagic stroke and had a NIHSSTOT
score. To focus the study, we chose to exclusively examine ischemic
stroke patients. Additionally, 2 of the patients had heights below 95 cm
and were dropped from the study. This selection reduced the initial data
set from 400 to 386 patients.</p>
<details>
<summary>
Filter Data
</summary>
<pre class="r"><code>#Build LR data set
LR.data = patients %&gt;%
  filter(!is.na(NIHSSTOT) &amp; IEESTRTY == &quot;ISC&quot; &amp; !(SubjectID %in% c(&quot;00272-014&quot;,&quot;00122-001&quot;)) )%&gt;%# Focus on ISC candidates and filter out patients with na HIHSSTOT 
  dplyr::select(SubjectID, NIHSSTOT # ID and response variables
                , GLUC, WBC, RBC, HCT, HBG  # Lab metrics from blood
                , CLCLTAR, CLCLTWT# Clot metrics
                , AGE, HEIGHT, WEIGHT, BMI # Numeric Demographics
                , MHNONE, MHPSIS, MHPSTIA, MHDVT, MHDM, MHHTN, SEX, SHMRJYN, SHALCYN # Binary
                , MHTHROMB, MHATEXCR # Quartary
                , SHALCUSE, MHPSISEL # Quintary
                ) </code></pre>
</details>
</div>
<div id="imputation" class="section level4">
<h4>Imputation</h4>
<p>Once the data of interest was selected, there were still a fair
amount of missing values in the data. For Numeric data, missing values
were replaced with the median value. Missing values for categorical
variables were encoded as ‘UNK’ for unknown.</p>
<details>
<summary>
Impute Median
</summary>
<pre class="r"><code># For Numeric Data, impute the median into missing values
for(i in 1:11){
  LR.data[is.na(LR.data[ , (i+2)]) , (i+2)] = median(LR.data[ , (i+2)][[1]], na.rm = T)
  }</code></pre>
</details>
</div>
<div id="power-transformations" class="section level4 tabset">
<h4 class="tabset">Power Transformations</h4>
<p>Power transformations were reviewed for each numeric variable that
showed signs of skewness or outliers. The following transformations were
made to the data to address skewness and reign in outliers.</p>
<div id="gluc-1" class="section level5">
<h5>GLUC</h5>
<p>The log-likelihood curve from box cox analysis (shown below) was used
to select an inverse transformation for the glucose variable.</p>
<ul>
<li><span class="math inline">\(GLUC_{new} =
\frac{GLUC_{old}^{-1}-1}{-1}\)</span></li>
</ul>
<p>The box plots below show the improvements realized by the
transformation.</p>
<hr>
<img src = "Data/GRAPHICS/BXCX_GLUC_L.png" width = 600>
<hr>
<img src = "Data/GRAPHICS/BXCX_GLUC_TRANS.png" width = 600>
<hr>
</div>
<div id="wbc-1" class="section level5">
<h5>WBC</h5>
<p>The log-likelihood curve from box cox analysis (shown below) was used
to select a log transformation for the white blood cell count
variable.</p>
<ul>
<li><span class="math inline">\(WBC_{new} = log(WBC_{old})\)</span></li>
</ul>
The box plots below show the improvements realized by the
transformation.
<hr>
<img src = "Data/GRAPHICS/BXCX_WBC_L.png" width = 600>
<hr>
<img src = "Data/GRAPHICS/BXCX_WBC_TRANS.png" width = 600>
<hr>
</div>
<div id="clcltar-1" class="section level5">
<h5>CLCLTAR</h5>
<p>The log-likelihood curve from box cox analysis (shown below) was used
to select a log transformation for the clot area variable.</p>
<ul>
<li><span class="math inline">\(CLCLTAR_{new} =
log(CLCLTAR_{old})\)</span></li>
</ul>
<p>The box plots below show the improvements realized by the
transformation.</p>
<hr>
<img src = "Data/GRAPHICS/BXCX_CLCLTAR_L.png" width = 600>
<hr>
<img src = "Data/GRAPHICS/BXCX_CLCLTAR_TRANS.png" width =600>
<hr>
</div>
<div id="clcltwt-1" class="section level5">
<h5>CLCLTWT</h5>
<p>The log-likelihood curve from box cox analysis (shown below) was used
to select a log transformation for the clot weight variable.</p>
<ul>
<li><span class="math inline">\(CLCLTWT_{new} =
log(CLCLTWT_{old})\)</span></li>
</ul>
<p>The box plots below show the improvements realized by the
transformation.</p>
<hr>
<img src = "Data/GRAPHICS/BXCX_CLCLTWT_L.png" width = 600>
<hr>
<img src = "Data/GRAPHICS/BXCX_CLCLTWT_TRANS.png" width = 600>
<hr>
</div>
<div id="bmi-1" class="section level5">
<h5>BMI</h5>
<p>The log-likelihood curve from box cox analysis (shown below) was used
to select a negative square root transformation for the BMI
variable.</p>
<ul>
<li><span class="math inline">\(BMI_{new} =
\frac{BMI_{old}^{-1/2}-1}{-1/2}\)</span></li>
</ul>
<p>The box plots below show the improvements realized by the
transformation.</p>
<hr>
<img src = "Data/GRAPHICS/BXCX_BMI_L.png" width = 600>
<hr>
<img src = "Data/GRAPHICS/BXCX_BMI_TRANS.png" width = 600>
<hr>
</div>
</div>
<div id="feature-engineering" class="section level4">
<h4>Feature Engineering</h4>
<p>The SHALCUSE variable indicates how many drinks an individual has per
week. This variable was converted to a numeric variable to reflect that
as the value increases, so does the number of drinks consumed by the
patient. This was the only candidate variable that needed to be refined
using feature engineering.</p>
<pre class="r"><code>LR.data = LR.data %&gt;%
  mutate(
         SHALC = case_when(
            SHALCUSE == &quot;1DRINK&quot; ~ 1
           ,SHALCUSE == &quot;2DRINK&quot; ~ 2
           ,SHALCUSE == &quot;3TO5DRINK&quot; ~ 4
           ,SHALCUSE == &quot;GTE6DRINK&quot; ~ 7
           ,TRUE ~ 0 ))</code></pre>
</div>
</div>
<div id="variable-selection" class="section level3">
<h3>Variable Selection</h3>
<p>Variable selection was performed by considering all pairwise
interactions between the numeric and categorical candidate features and
selecting a subset that yielded strong model performance. This process
compared the resulting R squared value, BIC, and RSS of models created
using forward selection, backward selection, and sequential replacement.
The resulting values are shown in the charts below which were used to
determine that a linear model with 12 variables would likely explain 15
- 20 percent of the variability within stroke severity without
drastically increasing the model BIC. A model with 12 parameters created
using a sequential replacement process was selected as the final linear
regression model and further refined to increase interpretability.</p>
<hr>
<img src = "Data/GRAPHICS/LR_VAR_SELECT.png" width = 600>
<hr>
</div>
<div id="model-refinement" class="section level3 tabset">
<h3 class="tabset">Model Refinement</h3>
<div id="first-iteration" class="section level4">
<h4>First Iteration</h4>
<p>The parameters and diagnostic plots for the first iteration of the
final model are shown below. The diagnostic plots indicate that the
linear model fits the data reasonably well and that we can proceed with
refining the model. We observe that the CLCLTAR.tran:MHHTN terms are not
statistically significant. The first model refinement is to drop these
from the model.</p>
<hr>
<img src = "Data/GRAPHICS/LR_I_1_COEF.png">
<hr>
<img src = "Data/GRAPHICS/LR_I_1_RES_PLT.png">
<hr>
</div>
<div id="second-iteration" class="section level4">
<h4>Second Iteration</h4>
<p>Dropping the CLCLTAR.tran:MHHTN terms from the model does not
drastically impact the R squared value. Reviewing the model we find that
the statistically significant terms involving sex only indicate
differences for females. The model can be simplified by Coding SEX as an
indicator variable for female. The same can be done for the MHPSIS
variable to indicate if the patient had a previous ischemic stroke. This
was done to create the final model on the next tab.</p>
<hr>
<img src = "Data/GRAPHICS/LR_I_2_COEF.png">
<hr>
<img src = "Data/GRAPHICS/LR_I_2_RES_PLT.png">
<hr>
</div>
<div id="final-model" class="section level4">
<h4>Final Model</h4>
<p>The final model explains 16.5 percent of the variability in the
stroke severity score among patients. The amount of variance explained
compared to the total variance is visually represented using histograms
on the following tab. The model has revealed some interesting details
regarding the explanatory variables. The final model parameters
estimates along with their p-values are shown below. The diagnostic
plots are also provided and indicate that the final model fits the data
reasonably well. Model interpretations are provided in the following
section.</p>
<hr>
<img src = "Data/GRAPHICS/LR_I_3_COEF.png">
<hr>
<img src = "Data/GRAPHICS/LR_I_3_RES_PLT.png">
<hr>
</div>
<div id="final-model-histograms" class="section level4">
<h4>Final Model Histograms</h4>
<p>The histogram below shows the distribution of the stroke severity
score among patients. The wide spread illustrates the variance in the
score.</p>
<hr>
<img src = "Data/GRAPHICS/LR_I_3_HIST_1.png">
<hr>
<p>The histogram below shows the distribution of the stroke severity
scores predicted by the final model. The narrow spread illustrates the
variance in predicted values. When we compare it against the first
histogram, we observe the the model predictions are more normal in
distribution and have a lower amount of variation. This depicts the
amount of variation in stroke severity that the final model accounts
for.</p>
<hr>
<img src = "Data/GRAPHICS/LR_I_3_HIST_2.png">
<hr>
</div>
</div>
<div id="model-interpretation" class="section level3">
<h3>Model Interpretation</h3>
<div id="intercept" class="section level4">
<h4>Intercept</h4>
<p>The intercept in this model can be interpreted as the expected stroke
severity score for male patients that:</p>
<ul>
<li>are not diabetic</li>
<li>don’t have a prior ischemic stroke</li>
<li>don’t have hypertension</li>
<li>have not used marijuana</li>
</ul>
</div>
<div id="age-1" class="section level4">
<h4>Age</h4>
<p>Age has a positive correlation with stroke severity score but it is
more statistically significant than practically significant. The
coefficient indicates that for each 13 year increase in age, the stroke
severity is expected to increase by 1 point.</p>
</div>
<div id="sex-interactions" class="section level4">
<h4>Sex Interactions</h4>
<p>If the patient is a female, her expected stroke severity score
depends on her Glucose levels, the size of the clot, her Hematocrit
(percent of red blood cells), and her Hemoglobin levels. The charts
below show how a female patient’s stroke severity score is expected to
be impacted by these levels. The gray and black lines indicate the 1st
quartile, median, and 3rd quartile for the glucose, clot area,
hematocrit, and hemoglobin among females and the range on the X axis
covers the max and min values measured in the study.</p>
<hr>
<img src = "Data/GRAPHICS/MODEL_INT_1.png">
<hr>
</div>
<div id="white-blood-cell-counts" class="section level4">
<h4>White Blood Cell Counts</h4>
<p>White blood cell count is positively correlated with stroke severity
for patients that have had a prior ischemic stroke and or have history
of using marijuana. The chart below shows how the stroke severity is
impacted by white blood cell count. The green line indicates the trend
for patients who have had a previous ischemic stroke but have not used
marijuana previously. The blue line shows the trend for patients who
have used marijuana previously and have had a prior ischemic stroke. The
red line shows the trend for patients who have both had a previous
ischemic stroke and have used marijuana in the past. The gray and black
lines show the 1st quartile, the median, and the 3rd quartile.</p>
<hr>
<img src = "Data/GRAPHICS/MODEL_INT_2.png">
<hr>
</div>
<div id="alcohol-consumption" class="section level4">
<h4>Alcohol Consumption</h4>
<p>Alcohol consumption was another factor that had an influence on a
patients stroke severity but its influence depended on if the patient
had a previous ischemic stroke, if they were diabetic, and if they had
hypertension. The trends for each of these and the possible combinations
are shown in the graph below. Alcohol consumption generally is
associated with a decrease in stroke severity for patients without
diabetes. This should be taken lightly since the majority of patients
didn’t report how much alcohol they consumed. For those patients, their
alcohol consumption was coded as a 0 and, as the chart indicates, no
adjustment was made to their score.</p>
<hr>
<img src = "Data/GRAPHICS/MODEL_INT_3.png">
<hr>
</div>
</div>
</div>
<div id="proteomics-data" class="section level2">
<h2>Proteomics Data</h2>
<p>The proteomics data set was evaluated using a differential expression
analysis to search for biomarkers. This was done by regressing the
NIHSSTOT response variable onto each protein using a linear model and
reviewing the resulting p-values to determine if the protein was
statistically significant. Two inherent challenges in this process were
handling skewness in the protein data and accounting for false discovery
among the large number of models created. These challenges were
addressed using the following approaches:</p>
<ol style="list-style-type: decimal">
<li>Skewness: The protein data was highly positively skewed with large
concentrations of 0s. This was addressed by treating 0s as missing
values for the proteins and then log transforming the non-zero values
prior to creating a linear model.</li>
<li>False Discovery: After creating a model for each protein, their
corresponding p-values were replaced with q-values using the approach
outlined in the Storey &amp; Tibshirani (2003) paper entitled
“Statistical significance for genomewide studies.”</li>
</ol>
<p>As is sometimes the case, none of the q-values associated with the
proteins were statistically significant. The search was expanded by
adding the candidate predictors from the hospital data as covariates in
the linear protein models. Additionally, new variable called STROKE_BELT
was introduced to see if any biomarkers existed when the model accounted
for regions in the US that have higher concentrations of strokes. This
wider sweep of the data identified 1 biomarker which was present in the
model that incorporated the individual’s sex.</p>
<p>The p-value plots below show the histograms of the p-values for the
base models, the models that incorporate the patient’s sex, and the
models that incorporate the stroke belt indicator. If there are no
significant proteins, then we expect the p-values to be uniformly
distributed. The chart below each histogram shows the proportion of
truly null values as a function of the tuning parameter <span
class="math inline">\(\lambda\)</span>. The cubic spline fit to the
<span class="math inline">\(\pi(\lambda)\)</span> vs <span
class="math inline">\(\lambda\)</span> is used to estimate the
proportion of null values at <span class="math inline">\(\lambda
=1\)</span> for determining the q-values.</p>
<div id="p-value-plots" class="section level3 tabset">
<h3 class="tabset">P Value Plots</h3>
<div id="base-model" class="section level4">
<h4>Base Model</h4>
<hr>
<img src = "Data/GRAPHICS/PROT_M1_PVALS.png">
<hr>
</div>
<div id="base-model-female" class="section level4">
<h4>Base Model + FEMALE</h4>
<hr>
<img src = "Data/GRAPHICS/PROT_M2_PVALS.png">
<hr>
</div>
<div id="base-model-stroke_belt" class="section level4">
<h4>Base Model + STROKE_BELT</h4>
<hr>
<img src = "Data/GRAPHICS/PROT_M3_PVALS.png">
<hr>
</div>
</div>
<div id="significant-proteins" class="section level3">
<h3>Significant Proteins</h3>
<p>The only proteins with statistically significant q-values where those
associated with protein batch 351 when the explanatory model includes
sex as a covariate:</p>
<ul>
<li><span class="math inline">\(NIHSSTOT = P_{351}+FEMALE\)</span></li>
</ul>
<p>The plot below shows what the resulting model looks like. When stroke
severity is regressed onto protein <span
class="math inline">\(P_{351}\)</span>, there is a statistically
significant relationship if sex is included as a covariate. Stroke
severity increases at the same rate for females and males but there is
an almost 10 point vertical shift between the lines indicating that
females have higher stroke severity if everything else is held constant.
Both genders experience higher stroke severity as the quantity of
protein <span class="math inline">\(P_{351}\)</span> increases.
Unfortunately, the data set is rather small and there are only 9
observations used in the model meaning the addition of a single point
that doesn’t fit the displayed trend could completely change the
results. This is due to treating 0s as missing values for the protein
data</p>
<hr>
<img src = "Data/GRAPHICS/SIG_PROT.png">
<hr>
<details>
<summary>
Model Details
</summary>
<hr>
<img src = "Data/GRAPHICS/PROT_M2_COEF.png">
<hr>
<img src = "Data/GRAPHICS/PROT_M2_RES.png">
<hr>
</details>
<p>Protein 351 represents 3 different proteins: P06493, Q07785, and
P61075. The majority protein is P06493. Details of each are provided
below.</p>
<ul>
<li>P06493
<ul>
<li>Protein: Cyclin-dependent kinase 1</li>
<li>Gene: CDK1</li>
<li>Function: “Plays a key role in the control of the eukaryotic cell
cycle by modulating the centrosome cycle as well as mitotic onset.”</li>
<li>Source: <a href="https://www.uniprot.org/uniprotkb/P06493/entry"
class="uri">https://www.uniprot.org/uniprotkb/P06493/entry</a></li>
</ul></li>
<li>Q07785
<ul>
<li>Protein: Cyclin-dependent kinase 2 homolog</li>
<li>Gene: CRK2</li>
<li>Function: “Involved in the control of cell cycle.”</li>
<li>Source: <a href="https://www.uniprot.org/uniprotkb/Q07785/entry"
class="uri">https://www.uniprot.org/uniprotkb/Q07785/entry</a></li>
</ul></li>
<li>P61075
<ul>
<li>Protein: Cyclin-dependent kinase 2 homolog</li>
<li>Gene: CRK2</li>
<li>Function: “Involved in the control of cell cycle.”</li>
<li>Source: <a href="https://www.uniprot.org/uniprotkb/P61075/entry"
class="uri">https://www.uniprot.org/uniprotkb/P61075/entry</a></li>
</ul></li>
</ul>
</div>
<div id="previously-linked-to-stroke-severity" class="section level3">
<h3>Previously linked to Stroke severity</h3>
<p>The proteins identified by our model are all Cyclin-Dependent Kinases
(CDKs) which have previously been linked to stroke cases. While our
model has failed to detect previously unknown biomarkers with regards to
stroke severity, it has produced some evidence to support previous
findings. The interested reader can learn more from the following
links:</p>
<ul>
<li>Science Direct - March 2002 - Cyclin-dependent kinases as potential
targets to improve stroke outcome: <a
href="https://www.sciencedirect.com/science/article/abs/pii/S0163725802001833"
class="uri">https://www.sciencedirect.com/science/article/abs/pii/S0163725802001833</a></li>
<li>National Library of Medicine - Cyclin-Dependent Kinases (CDK) and
Their Role in Diseases Development <a
href="https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7998717/"
class="uri">https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7998717/</a></li>
</ul>
</div>
</div>
<div id="snps" class="section level2">
<h2>SNPs</h2>
<p>The final data extracted as a part of this study were Single
Polynucleotide Polymorphisms (SNPs) which were collected for each of the
400 participants. SNPs represent pieces of human genetic code (DNA)
where substantial variability occurs and are particularly useful for
identifying disease causing genes. These data were evaluated using a
similar approach to that used for evaluating the proteomics data but had
their own unique set of challenges:</p>
<ol style="list-style-type: decimal">
<li>Data: The SNP data was spread across 6 different files which totaled
nearly 10 GB of data. Additionally, the data was not stored in a model
friendly format. Each file had to be wrangled into a format that could
be passed into a model and the results needed to be combined into a
single complete source of data. It took approximately 30 minutes of
computer run time to extract and transform the details required. The
result was a 1/2 GB file with 1 row for each patient (400) and one
column for each SNP (654K)</li>
<li>False Discovery: Similar to the proteomics data, 1 model had to be
created for each SNP which resulted in 654K p-values. False discovery
among these values was controlled by replacing p-values with q-values
using the approach outlined in the Storey &amp; Tibshirani (2003) paper
entitled “Statistical significance for genomewide studies.”</li>
<li>Computational Restrictions: Each model form considered had to be
created 654K times (once for each SNP). Depending on the model
complexity, this took 1 - 2 hours to create all 654K models for the
selected format. As a result, models of interest had to be carefully
selected based on the previously gathered details.</li>
</ol>
<div id="linear-regression-1" class="section level3">
<h3>Linear Regression</h3>
<p>The first set of models evaluated on the SNP data were linear
regression models where the stroke severity (NIHSSTOT) was used as the
response and the SNP was used as the predictor along with the hospital
data as covariates. P-values for each model were extracted using ANOVA
tests to determine statistical relevance of a given SNP. Unfortunately,
The histograms of the p-values of the resulting models are uniformly
distributed which is a key indication that none of the SNPs are truly
statistically significant. This was confirmed by calculating the
q-values which resulted in no statistically significant SNPs.</p>
<div id="nihsstot" class="section level4 tabset">
<h4 class="tabset">NIHSSTOT</h4>
<div id="base-model-1" class="section level5">
<h5>Base Model</h5>
<ul>
<li><strong>Forumula:</strong> LM1 = lm( formula = NIHSSTOT ~ SNP , data
= SNP_MODEL_DATA)</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_1.png">
<hr>
</div>
<div id="base-female" class="section level5">
<h5>Base + FEMALE</h5>
<ul>
<li><strong>Forumula:</strong> LM2 = lm( formula = NIHSSTOT ~ SNP +
FEMALE , data = SNP_MODEL_DATA)</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_2.png">
<hr>
</div>
<div id="base-prev_isc" class="section level5">
<h5>Base + PREV_ISC</h5>
<ul>
<li><strong>Forumula:</strong> LM3 = lm( formula = NIHSSTOT ~ SNP +
PREV_ISC , data = SNP_MODEL_DATA)</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_3.png">
<hr>
</div>
<div id="base-shmrjyn" class="section level5">
<h5>Base + SHMRJYN</h5>
<ul>
<li><strong>Forumula:</strong> LM4 = lm( formula = NIHSSTOT ~ SNP +
SHMRJYN , data = SNP_MODEL_DATA)</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_4.png">
<hr>
</div>
<div id="base-mhdmy" class="section level5">
<h5>Base + MHDMY</h5>
<ul>
<li><strong>Forumula:</strong> LM5 = lm( formula = NIHSSTOT ~ SNP +
MHDMY , data = SNP_MODEL_DATA)</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_5.png">
<hr>
</div>
</div>
</div>
<div id="logistic-regression" class="section level3">
<h3>Logistic Regression</h3>
<p>To widen the SNP analysis, a new response was selected. Some regions
of the US are prone to higher stroke rates and are said to reside within
the “stroke belt.” Each patient’s hospital location was known and the
binary variable STROKE_BELT was created to indicate if the patient
resided in the stroke belt. Logistic regression was used to model the
STROKE_BELT variable against the SNP data with the hospital data as
covariates. P-values for each logistic regression model were extracted
using ANOVA tests to determine statistical relevance of a given SNP. The
histograms of the p-values of the resulting models do appear to have
more values concentrated around 0 (a good sign) but when the
corresponding q-values were calculated, there were no statistically
significant results.</p>
<div id="stroke-belt" class="section level4 tabset">
<h4 class="tabset">STROKE BELT</h4>
<div id="base-model-2" class="section level5">
<h5>Base Model</h5>
<ul>
<li><strong>Forumula:</strong> LGR1 = glm( formula = STROKE_BELT ~ SNP ,
data = SNP_MODEL_DATA, family = binomial(link = “logit”))</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_6.png">
<hr>
</div>
<div id="base-model-female-1" class="section level5">
<h5>Base Model + FEMALE</h5>
<ul>
<li><strong>Forumula:</strong> LGR2 = glm( formula = STROKE_BELT ~ SNP +
FEMALE , data = SNP_MODEL_DATA, family = binomial(link = “logit”))</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_7.png">
<hr>
</div>
<div id="base-model-prev_isc" class="section level5">
<h5>Base Model + PREV_ISC</h5>
<ul>
<li><strong>Forumula:</strong> LGR3 = glm( formula = STROKE_BELT ~ SNP +
PREV_ISC , data = SNP_MODEL_DATA, family = binomial(link =
“logit”))</li>
</ul>
<hr>
<img src = "Data/GRAPHICS/SNP_HIST_8.png">
<hr>
</div>
</div>
</div>
</div>
<div id="summary" class="section level2">
<h2>Summary</h2>
<p>This analysis tackled 3 different sources of data with the broad
objective of finding any statistically significant details related to
stroke severity. It focused on creating explanatory models and used an
evidence focused approach. Evaluation of the hospital data resulted in a
model that helped explain the difference in stroke severity variance
between males and females. The model was able to explain approximately
16.5% of the variation in the entire data set. Analysis of the
proteomics data again found that there are statistically significant
differences in the stroke severity experienced by male and female
patients and found evidence to back previous medical studies that
connected Cyclin-Dependent Kinases (CDKs) to ischemic strokes. Finally,
this study evaluated SNP data with the hopes of identifying genes that
may cause strokes or that increase stroke severity and found that there
was no evidence of either for the patients in the study.</p>
</div>
</div>



</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
